docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+
+***Readonly auto-properties*** provide a more concise syntax to create
+immutable types. The closest you could get to immutable types
+in earlier versions of C# was to declare private setters:
+
+[!code-csharp[ClassicReadOnlyAutoProperty](../../samples/snippets/csharp/new-in-6/oldcode.cs#ClassicReadOnlyAutoProperty)]
+ 
+Using this syntax, the compiler doesn't ensure that the type really is immutable. It only
+enforces that the `FirstName` and `LastName` properties are not modified from any
+code outisde the class.
+
+Readonly auto-properties enable true readonly behavior. You declare the auto-property
+with only a get accessor:
+
+[!code-csharp[ReadOnlyAutoProperty](../../samples/snippets/csharp/new-in-6/newcode.cs#ReadOnlyAutoProperty)]
+
+The `FirstName` and `LastName` properties can be set only in the body of a constructor:
+
+[!code-csharp[ReadOnlyAutoPropertyConstructor](../../samples/snippets/csharp/new-in-6/newcode.cs#ReadOnlyAutoPropertyConstructor)]
+
+Trying to set `LastName` in another method generates a `CS0200` compilation error:
+
+```csharp
+public class Student
+{
+    public string LastName { get;  }
+
+    public void ChangeName(string newLastName)
+    {
+        // Generates CS 0200: Property or indexer cannot be assigned to -- it is read only
+        LastName = newLastName;
+    }
+}
+```
+
+This features enables true language support for creating immutable types and using
+the more concise and convenient auto-property syntax.
+
+### Auto Property Initializers
+
+***Auto Property Initializers*** let you declare the initial value for
+an auto property as part of the property declaration.  In earlier versions,
+these properties would need to have setters and a class author would need
+to use that setter to initialize the data storage used by the backing
+field. Consider this class for a student that contains the name and a
+list of the student's grades:
+
+[!code-csharp[Construction](../../samples/snippets/csharp/new-in-6/oldcode.cs#Construction)]
+ 
+As this class grows, you may include other constructors. Each constructor
+needs to initialize this field, or you'll introduce errors.
+
+C# 6 enables you to assign an initial value for the storage used by an
+auto property in the auto property declaration:
+
+[!code-csharp[Initialization](../../samples/snippets/csharp/new-in-6/newcode.cs#Initialization)]
+
+The `grades` member is initialized where it is declared. That makes it
+easier to perform the initialization exactly once. The initialization
+is part of the property declaration, making it easier to equate the
+storage allocation with public interface for `Student` objects.
+
+Property Initializers can be used with read / write properties as well
+as read only properties, as shown below.
+
+[!code-csharp[ReadWriteInitialization](../../samples/snippets/csharp/new-in-6/newcode.cs#ReadWriteInitialization)]
+
+## Expression bodied function members
+
+Many of the members that we write consist of single return statements.
```

Comment:
I understood the concept by seeing the examples but somehow I feel this needs to be explained a bit better. 
Wouldn't this be easier to understand if we said something like "The body of a lot of members that we write consist of only one return statement that can be represented as an expression. You can reduce that syntax by writing an **expression bodied member** instead. It works for methods and read-only properties."